### DISCUSSION TITLE
set language on python?

### DISCUSSION THREAD

**Discussion by fuka-oni** [openai/whisper#1185] [Q&A] [answered]
How I can use the --language on python? `options = whisper.DecodingOptions(language="Portuguese")` are not working.

```
#@title <-- Rodar o whisper para transcrever:

import os
import whisper
from tqdm import tqdm

# Define the folder where the wav files are located
root_folder = "/content/wavs"
exit_folder = "/content/saida"

# Set up Whisper client
print("Loading whisper model...")
model = whisper.load_model("large")
print("Whisper model complete.")

# Get the number of wav files in the root folder and its sub-folders
print("Getting number of files to transcribe...")
num_files = sum(1 for dirpath, dirnames, filenames in os.walk(root_folder) for filename in filenames if filename.endswith(".wav"))
print("Number of files: ", num_files)

options = whisper.DecodingOptions(language="Portuguese")

# Transcribe the wav files and display a progress bar
with tqdm(total=num_files, desc="Transcribing Files") as pbar:
    for dirpath, dirnames, filenames in os.walk(root_folder):
        for filename in filenames:
            if filename.endswith(".wav"):
                filepath = os.path.join(dirpath, filename)
                result = model.transcribe(filepath, fp16=False, verbose=True)
                transcription = result['text']
                # Write transcription to text file
                filename_no_ext = os.path.splitext(filename)[0]
                with open(os.path.join(exit_folder, filename_no_ext + '.txt'), 'w') as f:
                    f.write(transcription)
                pbar.update(1)
```

**Comment by glangford** [comment, marked as answer]
Try this 
```python
                result = model.transcribe(filepath, language="pt", fp16=False, verbose=True)
```



Para transcrever em português, também gosto de utilizar estas opções:
`patience=2, beam_size=5`
Slower but often higher quality.

**Reply by hodanli** [reply]
Hi,

Where did you get this information from? Is there a guide, or did you figure it out by looking at the code? I'm really interested and want to understand this library better.

**Reply by glangford** [reply]
> I'm really interested and want to understand this library better.

@hodanli The main project page, the Whisper research paper, and searching past discussions are good places to start. I also recommend experimenting with the different model sizes with different content, and see how the results vary. And yes, reading the code is helpful too.

**Reply by Gloridust** [reply]
@glangford Thx！Really helped me!

**Reply by vanyabrucker** [reply]
The language handling is implement [here at line 143 in whisper/whisper/transcribe.py](https://github.com/openai/whisper/blob/c0d2f624c09dc18e709e37c2ad90c039a4eb72a2/whisper/transcribe.py#L143)

And the language keys (e.g. "en", "zh", "de") are here in [whisper/whisper/tokenizer.py](https://github.com/openai/whisper/blob/c0d2f624c09dc18e709e37c2ad90c039a4eb72a2/whisper/tokenizer.py#L10)

Happy coding :)

**Comment by ulatekh** [comment]
You created `options = whisper.DecodingOptions(language="Portuguese")` but didn't actually use it...

**Comment by kaidfetic** [comment]
I want do transcribe an audio file,but in my native language and i dont know how,it needs to be fast and accurate so im using whisper.if anyone can help itl be great
